### ISSUE TITLE
Crash with strm creation for folders with thousands of files

### ISSUE THREAD

**Issue by gorman42** [cguZZman/plugin.googledrive#57]
**Describe the bug**
I had this happening to me repeatedly, with Kodi x64 under Windows 10 x64. The machine Kodi was running on had 32GB of RAM.

I haven't pinpointed the exact number of files after which Kodi crashed but the logs, which I activated, stopped without an error. Kodi crashed and the log wasn't written to anymore. Which is why I'm not attaching it.

**To Reproduce**
Unless it's a problem specific of my machine, I think it should be enough to have a single folder containing thousands of video files.
In my case the folder had over 14 thousands files.

**Comment by cguZZman** [comment]
Sounds possible. Can you do a test in other machine? And let us know if the same happens.

**Comment by gorman42** [comment]
Thanks for looking into this.
I've just tried with an export of 10,000+ files on a different machine and it didn't give any error. Windows 10 x64, 8GB RAM, Kodi 17.6 32 bit.
Kind of inconclusive, I know.

**Comment by cguZZman** [comment]
I see you tested it with a 32bit version now, not 64bits as the one crashing. Can you try with 32bits now instead of x64? Are both Kodis the same version? Can you test it with the latest version 18.1? Thanks

**Comment by gorman42** [comment]
I'll try with x86 version on the 32GB machine and report back. I'll use 18.1.

**Comment by gorman42** [comment]
Ok, I've run an export with 10,000+ files on both Kodi x64 and Kodi x86. It didn't crash.

BUT

But since opening this issue I actually created alphabet subfolders (A, B, C, D, etc.) where I put the files, according to alphabetic order. I needed to proceed with the export and I devised this system to do it (creating 27 folders under the main one, alphabet letters plus three "thematic" ones and then exporting them one by one).

Considering you developed the plugin, maybe you are able to tell if that makes a difference. Theoretically, as far as I understand it, it shouldn't. Because inside the A, B, C, D folders there are other folders, and those folders, before, were under the main folder that I was trying to export.

To clarify, the previous situation did not have those 10,000 files under a single folder. They were anyway inside 255 different folders. Now those 255 folders are inside the alphabet folders. I hope it's clear.

In any case, I cannot reproduce now and I honestly don't know what else to try.
